### PR TITLE
Add support for source-binary input

### PR DESCRIPTION
**Pull request by lucperkins**
Some of our Actions already implement source-binary on their own. This bakes in support for that.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -300,7 +289,68 @@ export abstract class DetSysAction {
...
+      correlation: this.identity,
+      facts: this.facts,
+      timestamp: new Date(),
+      uuid: randomUUID(),
+    });
+  }
+
+  /**
+   * Fetches a file in `.xz` format from the URL determined by the `source-*`
+   * parameters (or `source-binary` if set). It then imports that file's
```
Comment (cole-h) [inline src/index.ts:330]: Can we reword this to be more "general" (such that the `source-binary` parenthetical is unnecessary)? Maybe something like:

> Unpacks the closure specified by the `source-*` parameters, imports the contents into the Nix store, and returns the path..........

It may even be worth it to completely ignore the existence of `source-*` in this doc comment and document the behavior of them in `fetchArtifact` which actually implements the `source-*` handling (AFAIU)?

File: src/index.ts [hunk 2]
```diff
@@ -367,64 +416,27 @@ export abstract class DetSysAction {
...
-    const paths = stdout.split(os.EOL);
-    const lastPath = paths.at(-2);
-    return `${lastPath}/bin/${bin}`;
-  }
-
-  /**
-   * Fetch an artifact, such as a tarball, from the URL determined by the `source-*`
-   * inputs and other factors.
+   * Fetch an artifact, such as a tarball, from the URL determined by the
+   * `source-*` inputs.
```
Comment (cole-h) [inline src/index.ts:421]: nit:
```suggestion
   * Fetch an artifact, such as a tarball, from the location determined by the
   * `source-*` inputs.
```

(since none of them aside from `source-url` are actually URLs, and `source-binary` doesn't point at a URL unlike e.g. `source-pr` pointing to `i.d.s/pr/#`)

**Review by cole-h — APPROVED** [review]
1 nit, but LGTM

File: src/index.ts (hunk 2 above)
Comment (lucperkins) [reply src/index.ts:421]: Will do a quick follow-up
